Labs 1465/bundler demo code patch

108356037 opened the pull request:
Modify demo env to Sepolia, and directly interact with pre-deployed accounts.
I've modified the `NonStandardAccount.sol` so  anyone can send tx to it.

Commit:
delete auto funding when deploying

Commit:
signature always pass, but cannot move fund > 2 wei

Commit:
modify onlyOwner to only the owner EOA

Commit:
send 1 wei instead of 1 gwei

Commit:
don't need to really sign now

Commit:
update to sepolia env

Commit:
increase gas fee just to be safe

Commit:
auto addDeposit for 0.01 ether in script since we are using sepolia

Commit:
update README.md

Commit:
delete the section of valid pre-deployed account (may cause issues on bundler)

Commit:
still need sign in test case if user modified NonStandardAccount.sol

Commit:
stackup doesn't have sepolia endpoint so delete it

NIC619 commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# ETH Taipei 2023 Account Abstraction Workshop
+\# ETH Taipei 2023 Account Abstraction Workshop
```
```suggestion
# ETH Taipei 2023 Account Abstraction Workshop
```

NIC619 commented on README.md:
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
+Hint:
+
+1. Comment out codes that will likely cause failure at Bundler's validation stage.  
+2. (Optional) The `_validateSignature` function of `NonStandardAccount.sol` isn't properly implemented yet, it doesn't do any signature check. Add a verification logic in the function so only wallet owner can use the account. 
```
No need to implement this here unless we are tasking them with implementing a standard account with knowledge from prior tasks.

NIC619 commented on README.md:
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
 
-### Generate UserOp payload for bundler
-*(prerequisite: environment needs python3 installed to run below script)*
+1. `0x6137A181E3657A5dfd4Ca97C5bB1d50B3AAdb127`
+2. `0xa729a76caadb6fdcD9c198cEd19b5D4e54bA0485`
```
```suggestion
1. `0x6137A181E3657A5dfd4Ca97C5bB1d50B3AAdb127` (Account accessing BANNED OPCODE)
2. `0xa729a76caadb6fdcD9c198cEd19b5D4e54bA0485` (Account accessing invalid Storage Slot)
```

pilagod commented on README.md:
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
+For this demo, we will interact with two pre-deployed 4337 accounts on Sepolia testnet:
 
-### Generate UserOp payload for bundler
-*(prerequisite: environment needs python3 installed to run below script)*
```
Q: Don't need python anymore?

108356037 replied:
Still need, I added the comment back!

NIC619 commented on contracts/bundler/NonStandardAccount.sol:
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
 
         // Should fail
-        uint256 shoudlNotPassCall = IERC20(0x87224F6D41DF6044ddd30a87bBdEeBc8c8CAc4f0).balanceOf(address(entryPoint()));
+        uint256 shoudlNotPassCall = IERC20(0xf531B8F309Be94191af87605CfBf600D71C2cFe0).balanceOf(address(entryPoint()));
```
Turn the hardcoded address into a constant, easier to read and maintain.

NIC619 commented on contracts/bundler/NonStandardAccount.sol:
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
 
-        bytes32 hash = userOpHash.toEthSignedMessageHash();
-        if (owner != hash.recover(userOp.signature)) return SIG_VALIDATION_FAILED;
+        // To prevent bad actors transferring all the balance
+        (, uint256 value, ) = abi.decode(userOp.callData[4:], (address, uint256, bytes));
+        if (value > 2 wei) {
+            revert("This account is for demo purpose, please don't transfer large funds out from this account...");
+        }
```
This check should happen in `execute`

NIC619 commented on contracts/bundler/NonStandardAccount.sol:
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
+            revert("This account is for demo purpose, please don't transfer large funds out from this account...");
+        }
+
+        // Returns 0 anyway so everyone can pass signature validation
+        // (Optional) TODO: Add some verification logic here so only wallet owner can use this account
```
No need for this

pilagod commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# ETH Taipei 2023 Account Abstraction Workshop
+\# ETH Taipei 2023 Account Abstraction Workshop
```
```suggestion
# ETH Taipei 2023 Account Abstraction Workshop
```
Accident? 🧐

108356037 replied:
Forgot to run `npm run format` 0.0

pilagod reviewed (dismissed):
Others LGTM

NIC619 commented on script/bundler/DeployAccount.s.sol:
```diff
@@ -13,6 +13,8 @@ contract DeployAccount is Script {
     address ownerAccount = vm.envAddress("ACCOUNT_OWNER_ADDR");
 
+    // Deploys contract NonStandardAccount at contracts/bundler/NonStandardAccount.sol
+    // then automatically deposits 0.01 ether to the account from deployer
```
nit: the ether is deposited to Entrypoint, not the account
```suggestion
    // then deposits 0.01 ether for the account from deployer
```

NIC619 commented:
This `NonStandardAccount` account is for demonstration of BANNED OPCODE and invalid Storage Slot, other stuffs added to the account are only distractions. Remove the import of `BaseAccount` and modify from account example in previous tasks. This helps the participants to focus on the diff between each task, instead of getting a completely different account with many added features all of a sudden.

Commit:
update README.md

Commit:
refine NonStandard account

Commit:
fix nit

Commit:
update test, nonce is ignored now

Commit:
seperate scripts between deploying NonStandardAccount and SimpleAccount

Commit:
update README.md

Commit:
fix nit

Commit:
fix conflict

Commit:
merge main and fix conflict

Commit:
update README.md: add Deploy SimpleAccount script

108356037 commented:
Fixed all codes suggested by Nic, `NonStandardAccount` is now more simple and just inherits the `IAccount` interface. The deploy script is also fixed to only deploy `SimpleAccontFactory` & `SimpleAccount` using official sample.

pilagod commented on contracts/bundler/NonStandardAccount.sol:
```diff
@@ -1,99 +1,59 @@
-import { BaseAccount } from "aa/core/BaseAccount.sol";
 import { UserOperation } from "aa/interfaces/UserOperation.sol";
-import { IEntryPoint } from "aa/interfaces/IEntryPoint.sol";
+import { SignatureAccount } from "../SignatureAccount.sol";
```
```suggestion
```
nit: Unused

pilagod commented on script/bundler/DeploySimpleAccount.s.sol:
```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+/* solhint-disable avoid-low-level-calls */
+/* solhint-disable no-inline-assembly */
+/* solhint-disable reason-string */
```
```suggestion
/* solhint-disable avoid-low-level-calls no-inline-assembly reason-string */
```

108356037 replied:
Fixed!

Commit:
fix nit
